addons/product/models/product_attribute_value.py

```diff
@@ -98,26 +93,35 @@ def write(self, values):
             self.env.invalidate_all()
         return res
 
-    @api.ondelete(at_uninstall=False)
-    def _unlink_except_used_on_product(self):
+    def get_error_message(self):
         for pav in self:
             if pav.is_used_on_products:
```

Comment:
maybe we can improve it a bit here as well?
```suggestion
       if any(self.mapped('is_used_on_products')):
```